### PR TITLE
Implement chat in uimap.js

### PR DESCRIPTION
**Pull request by Crepepepe**
I'm very interested in your project.
I want to contribute your project as possible as I can.

I implement chat in uimap.js for command.

### CODE REVIEW

File: src/client/uimap.js [hunk 1]
```diff
@@ -13,6 +14,45 @@ UIMap.initialize = async function() {
...
 
+  this.chat = new MapleInput({
+    x: 5,
+    y: 540,
+    width: 530,
+    color: '#000000',
+    background: '#ffffff',
+    height: 13
+  });
+  this.chat.input.style.zIndex = 4;
```
Comment (johncintron) [inline src/client/uimap.js]: You should not be setting the z index for input elements. Please have a look at https://github.com/johncintron/nodin/blob/master/client/all.css#L31. Z index of 4 is reserved for the cursor, as seen here https://github.com/johncintron/nodin/blob/master/src/client/uicommon.js#L41. This is to ensure that the cursor appears above the input element.

File: client/index.html [hunk 2]
```diff
@@ -7,7 +7,7 @@
   </head>
   <body>
     <div id="game-wrapper">
-      <canvas id="game" width="800" height="600"></canvas>
+      <canvas id="game" width="800" height="600" tabindex="-1"></canvas>
```
Comment (johncintron) [inline client/index.html]: Why is `tabindex` necessary?

File: src/client/uimap.js [hunk 3]
```diff
@@ -13,6 +14,45 @@ UIMap.initialize = async function() {
...
+  this.chat.addSubmitListener(() => {
+    const msg = this.chat.input.value;
+    this.chat.input.value = "";
+    if (msg[0] == '!') {
```
Comment (johncintron) [inline src/client/uimap.js]: `===` please.

File: src/client/uimap.js [hunk 4]
```diff
@@ -13,6 +14,45 @@ UIMap.initialize = async function() {
...
+      const command = msg.split(' ');
+      console.log(command);
+      switch (command[0]) {
+        case "!level":
```
Comment (johncintron) [inline src/client/uimap.js]: I recommend wrapping the case bodies inside braces. See https://github.com/airbnb/javascript#comparison--switch-blocks.

File: src/client/uimap.js [hunk 5]
```diff
@@ -13,6 +14,45 @@ UIMap.initialize = async function() {
...
+    color: '#000000',
+    background: '#ffffff',
+    height: 13
+  });
+  this.chat.input.style.zIndex = 4;
+  this.chat.addSubmitListener(() => {
+    const msg = this.chat.input.value;
+    this.chat.input.value = "";
+    if (msg[0] == '!') {
+      const command = msg.split(' ');
```
Comment (johncintron) [inline src/client/uimap.js]: I think this is a bit cleaner: `const [command, ...commandArgs] = msg.split(' ');`

File: src/client/uimap.js [hunk 6]
```diff
@@ -175,7 +217,7 @@ UIMap.doRender = function(camera, lag, msPerTick, tdelta) {
   });
 
   this.drawNumbers(hp, maxHp, mp, maxMp);
-
+
```
Comment (johncintron) [inline src/client/uimap.js]: Unnecessary whitespace?

**Comment by Crepepepe** [comment]
I changed everything you mentioned
